pkg/validate/selinux_linux.go

```diff
@@ -35,122 +36,144 @@ var _ = framework.KubeDescribe("SELinux", func() {
+			})
+		}
 
-				By("create container for security context Privileged is true")
-				containerID := createPrivilegedContainer(rc, ic, podID, podConfig, "container-with-isPrivileged-process-label-test-", privileged)
+		Context("when single pod sandbox is not privileged", func() {
+			BeforeEach(func() {
+				sandboxID, sandboxConfig = framework.CreatePodSandboxForContainer(rc)
+			})
 
-				By("start container")
-				startContainer(rc, containerID)
-				Eventually(func() runtimeapi.ContainerState {
-					return getContainerStatus(rc, containerID).State
-				}, time.Minute, time.Second*4).Should(Equal(runtimeapi.ContainerState_CONTAINER_RUNNING))
+			AfterEach(func() {
+				cleanupSandbox(rc, sandboxID)
+			})
 
-				By("check the Privileged container")
-				checkProcessLabel(rc, containerID, privileged)
+			sandboxTests(false)
+		})
 
-				By("create pod")
-				privileged = false
-				podID, podConfig = createPrivilegedPodSandbox(rc, privileged)
+		Context("when single pod sandbox is privileged", func() {
+			BeforeEach(func() {
+				sandboxID, sandboxConfig = createPrivilegedPodSandbox(rc, true)
+			})
 
-				By("create container for security context Privileged is true")
-				containerID = createPrivilegedContainer(rc, ic, podID, podConfig, "container-with-notPrivileged-process-label-test-", privileged)
+			AfterEach(func() {
+				cleanupSandbox(rc, sandboxID)
+			})
+
+			sandboxTests(true)
+		})
+
+		Context("when multiple pod sandboxes are not privileged", func() {
+			var sandboxID2 string
+			var sandboxConfig2 *runtimeapi.PodSandboxConfig
+
+			BeforeEach(func() {
+				sandboxID, sandboxConfig = framework.CreatePodSandboxForContainer(rc)
+				sandboxID2, sandboxConfig2 = framework.CreatePodSandboxForContainer(rc)
+			})
 
-				By("start container")
-				startContainer(rc, containerID)
-				Eventually(func() runtimeapi.ContainerState {
-					return getContainerStatus(rc, containerID).State
-				}, time.Minute, time.Second*4).Should(Equal(runtimeapi.ContainerState_CONTAINER_RUNNING))
+			AfterEach(func() {
+				cleanupSandbox(rc, sandboxID)
+				cleanupSandbox(rc, sandboxID2)
+			})
 
-				By("check the Privileged container")
-				checkProcessLabel(rc, containerID, privileged)
+			SELIt("should create containers with different process labels", func() {
+				containerID := createContainerWithSelinux(rc, ic, sandboxID, sandboxConfig, nil, false, true, true)
+				containerID2 := createContainerWithSelinux(rc, ic, sandboxID2, sandboxConfig2, nil, false, true, true)
+				label1 := checkProcessLabelMCS(rc, containerID, false)
+				label2 := checkProcessLabelMCS(rc, containerID2, false)
+				Expect(label1).NotTo(Equal(label2))
 			})
 		})
-	}
+	})
 })
 
-func createContainerWithSelinux(rc internalapi.RuntimeService, ic internalapi.ImageManagerService, sandboxID string, sandboxConfig *runtimeapi.PodSandboxConfig, options *runtimeapi.SELinuxOption, shouldStart, shouldCreate bool) string {
+func SELIt(description string, doTest func()) {
```

Comment:
What's the reason of moving this check required for every test case?

Reply:
The main reason is because if selinux is not available the tests will not be registered, even as skipped. 

I do not think the current solution is ideal, I can revert it if desired, but it would be nice to see the tests are skipped rather than ignored and have a way to run them on non-selinux systems.

Reply:
Reverted this change to the old behavior to help move the PR through.